racksdb/web/ui/src/components/InfrastructurePopUp.vue

```diff
@@ -1,98 +1,41 @@
-<!--Copyright (c) 2022-2023 Rackslab
+<!--Copyright (c) 2022-2024 Rackslab
 
 This file is part of RacksDB.
 
 SPDX-License-Identifier: GPL-3.0-or-later -->
 
 <script setup lang="ts">
-import { ref } from 'vue'
-import type { Ref, PropType } from 'vue'
+import type { PropType } from 'vue'
 import type {
-  Infrastructure,
   NodeEquipment,
   NetworkEquipment,
-  StorageEquipment
+  StorageEquipment,
+  MiscEquipment
 } from '@/composables/RacksDBAPI'
-
-var showPopUp = ref(false)
-var popUpContent: Ref<NodeEquipment | NetworkEquipment | StorageEquipment | undefined> = ref()
-type equipmentType = 'nodes' | 'storage' | 'network'
-
-function popUp(name: String, equipment: equipmentType) {
-  if (showPopUp.value) {
-    showPopUp.value = !showPopUp.value
-  } else {
-    var layout = props.infrastructure.layout
-
-    for (var y = 0; y < layout.length; y++) {
-      var result = layout[y][equipment].find((item) => item.name === name)
-      if (result) {
-        popUpContent.value = result
-        break
-      }
-    }
-    showPopUp.value = true
-  }
-}
-
-function closePopUp() {
-  showPopUp.value = !showPopUp.value
-}
-
-const props = defineProps({
-  rack: String,
-  equipment: {
-    type: String as PropType<equipmentType>,
-    required: true
-  },
-  name: {
-    type: String,
-    required: true
-  },
-  id: String,
-  infrastructure: {
-    type: Object as PropType<Infrastructure>,
+import { Dialog, DialogPanel } from '@headlessui/vue'
+
+defineProps({
+  showPopUp: Boolean,
+  popUpContent: {
+    type: Object as PropType<
+      NodeEquipment | NetworkEquipment | StorageEquipment | MiscEquipment | undefined
```

Comment:
The complete equipment object is not needed in this component, only its type is required. You should better change this to restrict the manipulation on the type object only.